### DISCUSSION TITLE
ep 0 out was already available

### DISCUSSION THREAD

**Discussion by geekbozu** [hathach/tinyusb#962] [Q&A]
I am working on a custom vendor endpoint application using Tinyusb on an RP2040  to emulate an existing device. I am getting ep 0 out was already available when trying to use the existing application. Which then complains about being un-able to reset the usb device. Which makes sense as TinyUsb has panic'd on the above. 

I am confused as to what the driver is doing that is causing this error to happen. It is sending the 
"Set Clear and Halt" operation. Which seems to not be handled correctly... maybe because I am supposed to be handling it?

I see that there was a similar issue with webusb #609 however I am not sure how, Or even if that applies to the vendor endpoint as well. Just see that it was the same panic. 

If I had to guess I am not handling resetting properly in the tud_vendor_control_xfer_cb() ... however I'm not really sure what I should be doing? 
Am I supposed to be capturing the clear request and calling vendor_reset. Or am I supposed to be capturing something else?

My Code for this project is available here
https://github.com/geekbozu/PicoSilverLink/blob/main/src/SilverLink.c
My USB log is below from TinyUsb, 

[log.txt](https://github.com/hathach/tinyusb/files/6816264/log.txt)

**Comment by geekbozu** [comment]
So I tried closing all appropriate endpoints when SET CLEAR is called. 
As well as closing Endpoint 0 When tud_umount_cb() is called. Still crashes the same. 
[log.txt](https://github.com/hathach/tinyusb/files/6816259/log.txt)

**Comment by hathach** [comment]
please put all your log file as attached txt for readability. Make sure you use latest tinyusb instead of one included by pico-sdk

**Reply by geekbozu** [reply]
[log.txt](https://github.com/hathach/tinyusb/files/6816252/log.txt)
Sure here is the last log in text for you, 
I pulled this repo over the SDK repo it was built with commit ecb100a62fa99335fc41dca3d7c85a3903a7df65
